pwndbg/gdblib/regs.py

```diff
@@ -88,26 +89,29 @@ class module(ModuleType):
     last: Dict[str, int] = {}
 
     @pwndbg.lib.cache.cache_until("stop", "prompt")
-    def __getattr__(self, attr: str) -> int | None:
-        attr = attr.lstrip("$")
+    def read_reg(self, reg: str, frame: gdb.Frame | None = None) -> int | None:
+        reg = reg.lstrip("$")
         try:
-            value = gdb_get_register(attr)
-            if value is None and attr.lower() == "xpsr":
-                value = gdb_get_register("xPSR")
+            value = gdb_get_register(reg, frame)
+            if value is None and reg.lower() == "xpsr":
+                value = gdb_get_register("xPSR", frame)
             if value is None:
                 return None
             size = pwndbg.gdblib.typeinfo.unsigned.get(
                 value.type.sizeof, pwndbg.gdblib.typeinfo.ulong
             )
             value = value.cast(size)
-            if attr == "pc" and pwndbg.gdblib.arch.name == "i8086":
+            if reg == "pc" and pwndbg.gdblib.arch.name == "i8086":
                 if self.cs is None:
                     return None
                 value += self.cs * 16
             return int(value) & pwndbg.gdblib.arch.ptrmask
         except (ValueError, gdb.error):
             return None
 
+    def __getattr__(self, attr: str) -> int | None:
+        return self.read_reg(attr)
```

Comment:
Can we just do: `__getattr__ = self.read_reg`?

Reply:
Would have to be `__getattr__ = read_reg` but also mypy hates it and won't typecheck attribute lookups properly.